state-chain/pallets/cf-broadcast/src/tests.rs

```diff
@@ -1586,3 +1586,74 @@ fn aborted_broadcast_is_cleaned_up_on_success() {
 		assert!(AbortedBroadcasts::<Test, Instance1>::get().is_empty());
 	});
 }
+
+#[test]
+fn threshold_sign_and_broadcast_with_historical_key_full_flow() {
```

Comment:
We probably also want to test that we don't penalise validators in case they fail?